astropy/io/fits/header.py

```diff
@@ -2047,9 +2045,7 @@ def __get__(self, obj, owner=None):
                 obj._header_str = None
             else:
                 raise AttributeError(
-                    "'{}' object has no attribute '_header'".format(
-                        obj.__class__.__name__
-                    )
+                    f"'{obj.__class__.__name__}' object has no attribute '_header'"
```

Comment:
This could be edited more aggressively:
```suggestion
                    f"'{type(obj).__name__}' object has no attribute '_header'"
```

Reply:
Done